code/controllers/subsystem/throwing.dm

```diff
@@ -195,6 +197,6 @@ SUBSYSTEM_DEF(throwing)
 		var/atom/movable/AM = thing
 		if (AM == thrownthing || (AM == thrower && !ismob(thrownthing)))
 			continue
-		if (AM.density && !(AM.pass_flags & LETPASSTHROW) && !(AM.flags_1 & ON_BORDER_1))
+		if ((AM.density || (isliving(AM) && can_hit_nondense_mobs)) && !(AM.pass_flags & LETPASSTHROW) && !(AM.flags_1 & ON_BORDER_1))
```

Comment:
This is an exception for a very specific case, which can have unintended consequences. For example, you'll be able to hit jaunting things and abstract/ethereal living which shouldn't be.

Add a variable at the atom level, something like `throw_cover_percentage`, and then run something like `AM.density || prob(AM.throw_cover_percentage)`

Or something like that which we can expand upon.

Reply:
That would make it a property of the thing being thrown at, rather than the thing being thrown which is what I need. The check for `&& !(AM.pass_flags & LETPASSTHROW)` lets nondense things that REALLY don't want to be hit with throws forcibly opt out, if any such mobs exist. I thought being able to bonk a revenant with a heater would be funny, but I've added LETPASSTHROW to their passflags so that won't be the case. 

As for jaunting or shadow walking things specifically, while jaunting the mob is moved to an `/obj/effect/dummy/phased_mob/spell_jaunt` which is an effect and not a living mob, so we couldn't hit them anyway,